helper/resource/teststep_validate_test.go

```diff
@@ -79,12 +80,27 @@ func TestTestStepValidate(t *testing.T) {
 		testStepValidateRequest testStepValidateRequest
 		expectedError           error
 	}{
-		"config-and-importstate-missing": {
-			testStep: TestStep{},
+		"config-and-importstate-and-refreshstate-missing": {
+			testStep:                TestStep{},
+			testStepValidateRequest: testStepValidateRequest{},
+			expectedError:           fmt.Errorf("TestStep missing Config or ImportState or RefreshState"),
+		},
+		"refreshstate-first-step": {
+			testStep: TestStep{
+				RefreshState: true,
+			},
 			testStepValidateRequest: testStepValidateRequest{
-				TestCaseHasProviders: true,
+				StepNumber: 1,
 			},
-			expectedError: fmt.Errorf("TestStep missing Config or ImportState"),
+			expectedError: fmt.Errorf("TestStep cannot have RefreshState as first step"),
+		},
+		"importstate-and-refreshstate-both-true": {
+			testStep: TestStep{
+				ImportState:  true,
+				RefreshState: true,
+			},
+			testStepValidateRequest: testStepValidateRequest{},
+			expectedError:           fmt.Errorf("TestStep cannot have ImportState and RefreshState in same step"),
 		},
```

Comment:
Just to ensure coverage of new validation:

```suggestion
		},
		"destroy-and-refreshstate-both-true": {
			testStep: TestStep{
				Destroy:      true,
				RefreshState: true,
			},
			testStepValidateRequest: testStepValidateRequest{},
			expectedError:           fmt.Errorf("TestStep cannot have RefreshState and Destroy"),
		},
```